quinn-proto/src/crypto/rustls.rs

```diff
@@ -157,7 +157,7 @@ impl crypto::Session for TlsSession {
     }
 
     fn write_handshake(&mut self, buf: &mut Vec<u8>) -> Option<Keys<Self>> {
-        let keys = self.write_hs(buf)?;
+        let secrets = self.write_hs(buf)?;
```

Comment:
@Ralith any suggestions for how to deal with this?

Reply:
Deal with what, exactly? At a glance it looks like the desired keys are available via `Secrets::packet_keys`, but maybe I'm missing something? We'd also want to hold onto the 1-RTT `secrets` for key update purposes.

Reply:
Mostly, where do we stick `Secrets`? Does it need yet another associated type on the `crypto::Session`?

Reply:
Can we store it inside `struct TlsSession` directly?

Reply:
Yeah, I think that might work. I forgot we had that, I remembered us having the `SessionKind` enum as the top level thing.